kannon/master.py

```diff
@@ -49,16 +49,17 @@ def build(self, root_task: gokart.TaskOnKart) -> None:
         task_queue = self._create_task_queue(root_task)
 
         # consume task queue
-        launched_task_ids: Set[str] = set()
         logger.info("Consuming task queue...")
         while task_queue:
             task = task_queue.popleft()
             if task.complete():
                 logger.info(f"Task {self._gen_task_info(task)} is already completed.")
                 continue
-            if task.make_unique_id() in launched_task_ids:
+            if task.make_unique_id() in self.task_id_to_job_name:
+                # check if task is still running on child job
+                assert self._check_child_task_status(task), f"Child task {self._gen_task_info(task)} failed."
```

Comment:
It would be better to keep the master node running while the child nodes are running. If we add ownerReference to child node, the running child nodes will be killed by Kubernetes. https://github.com/m3dev/kannon/pull/38

What do you think?

Reply:
@maronuu @kitagry I agree with @mamo3gr 

I think it's ok to kill the master (and other children with !38 ) for now.

I'm not sure how it's problematic to kill them because they may have gokart caches and easy to retry.

Let's add other option if we confirm any problem.